### PR TITLE
feat: Copy to Clipboard & Show Alert for the password being copied into the clipboard - WPB-4639

### PR DESCRIPTION
**Pull request by agisilaos**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4639" title="WPB-4639" target="_blank">TaskWPB-4639</a>  [iOS] Copy to clipboard
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Also fixes: [WPB-4640](https://wearezeta.atlassian.net/browse/WPB-4640)

With this PR I introduced the copy to clipboard function when the user taps on the Create Link Button. After that, we show an alert that the password has been copied to the clipboard. In addition to that I did a small refactoring:

#### Introduced CreateSecureGuestLinkPasswordValidatorHelper

In that class, we have two methods related to setting the color for certain elements based on the result of the password validation. 

1. func displayPasswordErrorState(for textFields: [UITextField], for labels: [UILabel])

2. func resetPasswordDefaultState(for textFields: [UITextField], for labels: [UILabel])

This is one of the ways that we can keep the View Controller as light as possible. 

#### Introduced a new file for the DownStyle extension

So I could move the code outside of the View Controller. 

#### Introduced a new method called `presentPasswordCopiedAlert`

Where it presents an alert with an OK button as an action.

Designs can be found [here](https://www.figma.com/file/46vAH5LlYhPpkgV4n1C89M/Password-protected-guest-links?type=design&node-id=0-1&mode=design&t=CYTW1iCPHjEeyWy5-0)

-- 

I'm planning to do some more refactoring but on a separate PR. 

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

[WPB-4640]: https://wearezeta.atlassian.net/browse/WPB-4640?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreateSecureGuestLink/CreateSecureGuestLinkViewController.swift [hunk 1]
```diff
@@ -122,14 +124,27 @@ class CreateSecureGuestLinkViewController: UIViewController, CreatePasswordSecur
...
+
+    private var createSecureGuestLinkPasswordValidatorHelper = CreateSecureGuestLinkPasswordValidatorHelper()
 
     // MARK: - Override methods
 
     override func viewDidLoad() {
         super.viewDidLoad()
         setUpViews()
         setupConstraints()
+        textFieldDidChange(securedGuestLinkPasswordTextfield)
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreateSecureGuestLink/CreateSecureGuestLinkViewController.swift:147]: why do we call this in viewDidLoad?

Maybe better to have a private method making the configuration and `textFieldDidChange` calling this method

**Review by netbe — APPROVED** [review]
just a small suggestion

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreateSecureGuestLink/CreateSecureGuestLinkViewController.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreateSecureGuestLink/CreateSecureGuestLinkViewController.swift:147]: The reason is for setting the state for the Create link button.

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreateSecureGuestLink/CreateSecureGuestLinkPasswordValidatorHelper.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    /// - Parameters:
+    ///   - textFields: Array of TextFields
+    ///   - labels: Array of Labels
+    func resetPasswordDefaultState(for textFields: [UITextField], for labels: [UILabel]) {
+        for textField in textFields {
+            textField.applyStyle(.default)
+
+            for label in labels {
+                label.textColor = defaultTextColor
+            }
```
Comment (David-Henner) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreateSecureGuestLink/CreateSecureGuestLinkPasswordValidatorHelper.swift]: is this for loop meant to be nested in the other one (the one above) ?

Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreateSecureGuestLink/CreateSecureGuestLinkPasswordValidatorHelper.swift]: No isn't meant to be nested. Good catch, I fixed it.

**Review by David-Henner — APPROVED** [review]

**Pull request merged**